### PR TITLE
gRPC support

### PR DESCRIPTION
**Pull request by y3llowcake**
First pass at gRPC support. Only one method implemented, if the approach looks good I can fill in the rest.

```┌─(cy@kudu:~/go/src/github.com/foursquare/quiver)                                              
└─(130:31)% go build && ./quiver --rpc-port=1234 --grpc-port=1337 a=./hfile/testdata/pairs.hfile
2019/04/13 11:03:59 Quiver version HEAD? (built unknown?, go1.12.2).                           

2019/04/13 11:03:59 Loading collections...     
2019/04/13 11:03:59 [Reader.NewReader] Reading in a (1.71mb)...                                
2019/04/13 11:03:59 [Reader.NewReader] Loaded a.                                               
2019/04/13 11:03:59 Serving on http://kudu:9999/                                               
2019/04/13 11:03:59 [9999]                     
2019/04/13 11:03:59 adminz registered          
2019/04/13 11:03:59 Watching paths for killfile: [/dev/shm/healthz/kill.9999 /dev/shm/healthz/kill.all]
2019/04/13 11:03:59 Listening for raw RPC on 1234                                              
2019/04/13 11:03:59 Listening for gRPC on 1337 
```

```
┌─(cy@kudu:~/go/src/github.com/foursquare/quiver)                            
└─(25)% grpc_cli call localhost:1337 foursquare.quiver.client.QuiverService/GetValuesSingle "hfile_name:'a' sorted_keys:'\0\0\0\1'"
connecting to localhost:1337          
values {           
  value: "~1"      
}                  
key_count: 1       

Rpc succeeded with OK status 
```

### CODE REVIEW

**Comment by slackhappy** [comment]
For some background, at Slack we only use the SingleHFileKeyRequest.  We use both thrift and grpc here.  This change would allow us to use a grpc sidecar on our webserving nodes for more stable connections.

If you have any concerns, we are happy to address them!  @jvandew

**Comment by jvandew** [comment]
holy cow this looks awesome, thank you!

I'll give this an in-depth read through in the next day or two if that works, and hopefully find someone with more go expertise than I have to take a look as well heh

**Comment by slackhappy** [comment]
Thanks Jacob!  Happy to provide a bit more context if it is helpful

**Review by jvandew — APPROVED** [review]
looks reasonable to me -- I am headed out on vacation tomorrow actually, but going to pull in @krrusso and @ghthor to get at least one set of experienced go eyes on this and land it as they see fit.

Thank you for this patch! I'd love to hear some more about your plans sometime 😄

File: rpc.go [hunk 1]
```diff
@@ -69,19 +92,49 @@ func (cs *ThriftRpcImpl) GetValuesSingle(req *gen.SingleHFileKeyRequest) (r *gen
...
 	if Settings.debug {
-		log.Printf("[GetValuesSingle] %s found %d of %d.\n", *req.HfileName, found, len(req.SortedKeys))
+		log.Printf("[GetValuesSingle] %s found %d of %d.\n", req.HfileName, found, len(req.SortedKeys))
 	}
-	res.KeyCount = &found
+	res.KeyCount = found
 	return res, nil
 }
 
+func (g *GrpcImpl) GetValuesSingle(_ context.Context, req *pb.SingleHFileKeyRequest) (*pb.SingleHFileKeyResponse, error) {
```
Comment (ghthor) [inline rpc.go:108]: Glad to see the intention to start threading context objects throughout all these request chains.

**Review by ghthor — APPROVED** [review]
Seems fine to me. Thanks for beginning this work.

**Comment by slackhappy** [comment]
Thanks for the review @ghthor !

**Pull request merged**